src/reactor/reactors.rs

```diff
@@ -162,9 +164,23 @@ impl Reactor {
             }
 
             Replies::ConditionalCheckFailed(cmd) => {
-                warn!("conditional check failed {:?}", cmd);
                 if writer.id.0 == cmd.writer_id {
-                    writer.fail_events_upon_conditional_check_failure(cmd.event_number);
+                    if writer.reconnect {
+                        // Note: it still could be caused by interleaved data, but the probability is low.
+                        // Right now there is no good way to tackle this issue. Only application side
+                        // can tell if this is a false alarm or not.
+                        warn!(
+                            "conditional check failed {:?}, probably a false alarm caused by reconnection",
+                            cmd
+                        );
+                        writer.ack(cmd.event_number);
+                        writer.reconnect = false;
```

Comment:
Based on the discussion here:
https://github.com/pravega/pravega-client-rust/issues/198#issuecomment-747818033
It seems like the thing to do is that if `reconnect` is `true` then treat `ConditionalCheckFailed` as a normal connection failure and trigger another re-connect.